packages/app/src/settings/device/NotificationSettings.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+  <SettingsSection>
+    <template #title>
+      {{ t('settingsPage.notifications.title') }}
+    </template>
+    <template #description>
+      {{ t('settingsPage.notifications.description') }}
+    </template>
+    <div class="divide-y border rounded divide-gray-50 border-gray-100 px-[16px]">
+      <div
+        v-for="({id, title}) in switches"
+        :key="id"
+        class="py-[16px]"
+      >
+        <h4 class="flex text-gray-800 text-[16px] leading-[24px] items-center">
+          <label :for="id">{{ title }}</label>
+          <Switch
+            class="mx-[8px]"
+            :value="props.gql.localSettings?.preferences[id] ?? false"
+            :name="id"
+            @update="(value) => updatePref(value, id)"
+          />
+          {{ gql.localSettings }}
+        </h4>
+      </div>
+      <div class="py-[16px]">
+        <h4 class="flex text-gray-800 text-[16px] leading-[24px] items-center">
+          {{ t('settingsPage.notifications.notifyMeWhenRunCompletes') }}
+        </h4>
+        <div class="pt-[5px]">
+          <Checkbox
+            v-for="({id, label}) in statuses"
+            :id="id"
+            :key="id"
+            v-model="listRef"
+          >
+            <template #label>
+              <label
+                :for="id"
+                class="text-sm text-gray-900"
+              >
+                {{ label }}
+              </label>
+            </template>
+          </Checkbox>
+        </div>
+      </div>
+      <div class="py-[16px]">
+        <Button
+          variant="white"
+          size="32"
+          @click="showTestNotification"
+        >
+          {{ t('settingsPage.notifications.sendTestNotification') }}
+        </Button>
+        <i18n-t
+          tag="p"
+          scope="global"
+          keypath="settingsPage.notifications.notReceivingNotifications"
+          class="pt-[15px] text-gray-600 text-sm font-normal"
+        >
+          <ExternalLink href="#">
+            {{ t('settingsPage.notifications.troubleshoot') }}
+          </ExternalLink>
+        </i18n-t>
+      </div>
+    </div>
+  </SettingsSection>
+</template>
+
+<script lang="ts" setup>
+// TODO: Add documentation link https://github.com/cypress-io/cypress-documentation/issues/5280
+import { ref, watch } from 'vue'
+import { gql, useMutation } from '@urql/vue'
+import { useI18n } from '@cy/i18n'
+import Switch from '@packages/frontend-shared/src/components/Switch.vue'
+import SettingsSection from '../SettingsSection.vue'
+import { NotificationSettingsFragment, SetNotificationSettingsDocument, NotificationSettings_ShowNotificationDocument } from '../../generated/graphql'
+import Checkbox from '@cy/components/Checkbox.vue'
```

Comment:
FYI - there is a [`Checkbox` available from the design system](https://design.cypress.io/components/vue/Checkbox.html). Is the intent to use our existing implementation?

Edit: DS version is restricted to `boolean` so maybe not a real option without some serious updates, but figured I should still ask the question

Reply:
I didn't realize that there was one... I don't think it fits this use case easily, so for now we should move forward with this I think. It looks like this checkbox component wasn't used anywhere else except its test before this